api/src/services/healthService.js

```diff
@@ -1,140 +1,261 @@
+}
+
+async function checkBlockchainHealth(req, res) {
+  const { query } = req;
+  let networks = splitParams(query.networks);
+  
+  const validationErrors = validateNetworks(networks);
+  if (validationErrors.length) {
+    return res.status(400).json({
+      msg: 'Validation failed',
+      errors: validationErrors
     });
   }
+
+  let healthErrors = [];
+  await Promise.all(networks.map(async network => {
+    try {
+      await cereNetworkService.getHealth(network);
+    } catch (error) {
+      console.error(error);
+      healthErrors.push(`Unable to connect to ${network} network`);
+    }
+  }));
+
+  res.status(200).json(
+    healthErrors.length
+      ? { msg: 'Health check failed', errors: healthErrors }
+      : { msg: `${networks.join(',')} status is ok`}
+  );
+}
+
+async function checkBlockchainBlocksFinalization(req, res) {
+  const { query } = req;
+  let networks = splitParams(query.networks);
+  const cereConfig = blockchains.find(blockchain => blockchain.name === blockchainNames.CERE);
+  
+  const validationErrors = validateNetworks(networks);
+  if (validationErrors.length) {
+    return res.status(400).json({
+      msg: 'Validation failed',
+      errors: validationErrors
+    });
+  }
+  
+  let healthErrors = [];
+  await Promise.all(
+    networks.map(async network => {      
+      const [best, finalized] = await Promise.all([
+        cereNetworkService.getFinalizedBlock(network),
+        cereNetworkService.getBestBlock(network)
+      ]);      
+      const bestBlockNumberDiff = best - finalized;
+      if (bestBlockNumberDiff > cereConfig.bestBlockNumberDiff) {
+        healthErrors.push(`${network} network best block finalization difference number ${bestBlockNumberDiff} below the ${cereConfig.bestBlockNumberDiff} threshold`);
+      }
+    })
+  );
+
+  res.status(200).json(
+    healthErrors.length 
+      ? { msg: 'Health check failed', errors: healthErrors }
+      : { msg: `${networks.join(',')} best block finalization difference number is ok`}
+  );
+}
+
+async function checkBlockchainBlocksProducing(req, res) {
+  const { query } = req;
+  let networks = splitParams(query.networks);
+  const cereConfig = blockchains.find(blockchain => blockchain.name === blockchainNames.CERE);
+  
+  const validationErrors = validateNetworks(networks);
+  if (validationErrors.length) {
+    return res.status(400).json({
+      msg: 'Validation failed',
+      errors: validationErrors
+    });
+  }  
+  
+  let healthErrors = [];
+  await Promise.all(
+    networks.map(async network => {
+      const { block } = await cereNetworkService.getLatestBlock(network);
+      const [previousTime, lastTime] = await Promise.all([
+        cereNetworkService.getBlockTime(network, block.header.number - 1),
+        cereNetworkService.getBlockTime(network, block.header.number),
+      ]);
+      const timeDiff = lastTime - previousTime;
+      if (timeDiff > cereConfig.blockTimeDiffMs) {
```

Comment:
`cereConfig.blockTimeDiffMs` should be updated in the k8s config?

Reply:
yes